realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -287,7 +287,11 @@ void RealSenseNodeFactory::init()
             }
             if (_device)
             {
-                startDevice();
+                bool rosbag_loop(declare_parameter("rosbag_loop", rclcpp::ParameterValue(false)).getrclcpp::PARAMETER_BOOL());
+                do
+                {
+                    startDevice();
+                } while (rosbag_loop); // Terminate loop only after CTRL-C
```

Comment:
Hi @acornaglia ,
Will changing this param to False on runtime stop this while true?
If not
Why not add a counter instead of bool to the parameter and loop by the counter?
while (true) is less preferred IMO..
You can add a special case of -1 for example for `forever`,
@SamerKhshiboun  thoughts?

Reply:
Hi @Nir-Az , thanks for the comment. My expectation is that the bag file will loop only when the node is launched with a command similar to the following:
```
ros2 run realsense2_camera realsense2_camera_node --ros-args -p rosbag_filename:="depth_under_water.bag" -p rosbag_loop:="true"
```
As such, I don't expect the `rosbag_loop` parameter changing at runtime.  It should be set initially based on the value provided in the command line. I am open to the idea of using a counter instead of a boolean, but could you please provide a more concrete example? Thank you.